pipeline/pipeline/calculate_metrics_file.py

```diff
@@ -81,27 +46,19 @@ def calculate_metrics(
     # print(drop_rows.index[0:5])
     df.drop(drop_rows.index, axis=0, inplace=True)
 
-    # create non imputed temp column before it gets imputed.Temp column is created just to store values
-    # It will be deleted and values transferred to another column to not affect the position of columns
-    df["delta.tested.non.impu.temp"] = df["delta.tested"].fillna(value=0)
-    df["delta.confirmed.non.impu.temp"] = df["delta.confirmed"].fillna(value=0)
-    df["delta.deceased.non.impu.temp"] = df["delta.deceased"].fillna(value=0)
-    df["delta.recovered.non.impu.temp"] = df["delta.recovered"].fillna(value=0)
-    df["delta.other.non.impu.temp"] = df["delta.other"].fillna(value=0)
-
-    # impute data
-    df["delta.tested"] = imputeCols(df["delta.tested"])
-    df["delta.confirmed"] = imputeCols(df["delta.confirmed"])
-    df["delta.deceased"] = imputeCols(df["delta.deceased"])
-    df["delta.recovered"] = imputeCols(df["delta.recovered"])
-    df["delta.other"] = imputeCols(df["delta.other"])
+    #  filling out na with 0 values
+    df["delta.tested"] = df["delta.tested"].fillna(value=0)
+    df["delta.confirmed"] = df["delta.confirmed"].fillna(value=0)
+    df["delta.deceased"] = df["delta.deceased"].fillna(value=0)
+    df["delta.recovered"] = df["delta.recovered"].fillna(value=0)
+    df["delta.other"] = df["delta.other"].fillna(value=0)
 
     # generate new "total" columns
-    df["total.confirmed"] = df["delta.confirmed"].cumsum()
-    df["total.tested"] = df["delta.tested"].cumsum()
-    df["total.recovered"] = df["delta.recovered"].cumsum()
-    df["total.deceased"] = df["delta.deceased"].cumsum()
-    df["total.other"] = df["delta.other"].cumsum()
+    df["total.confirmed"] = df["delta.confirmed"].cumsum().fillna(value=0)
+    df["total.tested"] = df["delta.tested"].cumsum().fillna(value=0)
+    df["total.recovered"] = df["delta.recovered"].cumsum().fillna(value=0)
+    df["total.deceased"] = df["delta.deceased"].cumsum().fillna(value=0)
+    df["total.other"] = df["delta.other"].cumsum().fillna(value=0)
```

Comment:
why not switch to just .fillna(..) here? the API provides us with totals... no need to recalculate them ourselves since we are no longer imputing anything.

Reply:
Will remove the totals in phase 2. Do not want to disturb the number of columns as that will affect the dashboard. Trying to keep the disturbance as little as possible.